app/controllers/Controller.scala

```diff
@@ -120,11 +121,22 @@ abstract class LoggedOutController()
   }
 }
 
+trait HeaderCarrierConversion
+  extends uk.gov.hmrc.play.bootstrap.controller.BaseController
+    with uk.gov.hmrc.play.bootstrap.controller.Utf8MimeTypes {
+
+  override implicit def hc(implicit rh: RequestHeader): HeaderCarrier =
+    HeaderCarrierConverter.fromHeadersAndSessionAndRequest(rh.headers, Some(rh.session), Some(rh))
+}
+
 abstract class BaseController()
-  extends AuthConfigImpl with I18nSupport with FrontendController with HeaderEnricher {
+  extends AuthConfigImpl with I18nSupport with HeaderCarrierConversion with HeaderEnricher {
 
   val errorHandler: ErrorHandler
   val sessionService: SessionService
+
+  implicit def ec: ExecutionContext
```

Comment:
Any reason this is not a val?

Reply:
Because it's abstract.

Reply:
Doesn't need to be a def though

Reply:
It should be defined as such when abstract in a trait. This allows someone to override it any way they want (including with lazy).  A val does not.